include/private/error.h

```diff
@@ -241,22 +415,74 @@ COLD_FUNCTION
 void
 raise_resolve_hostname_failure( const char *message );
 
+/**
+ * Raises an error indicating a socket bind failure.
+ *
+ * @param message   Localized description of the failure.
+ * 
+ * @param code      Error code from the failing call.
+ * 
+ * @param code_type Localized description of what the code represents.
+ *
+ * **Thread Safety: MT-Safe**
+ * **Async Signal Safety: AS-Unsafe**
+ * **Async Cancel Safety: AC-Unsafe**
+ */
 COLD_FUNCTION
 void
 raise_socket_bind_failure( const char *message,
                            int code,
                            const char *code_type );
 
+/**
+ * Raises an error indicating that connecting a socket failed.
+ *
+ * @param message   Localized description of the failure.
+ * 
+ * @param code      Error code from the failing call.
+ * 
+ * @param code_type Localized description of what @p code represents.
+ *
+ * **Thread Safety: MT-Safe**
+ * **Async Signal Safety: AS-Unsafe**
+ * **Async Cancel Safety: AC-Unsafe**
+ */
 COLD_FUNCTION
 void
 raise_socket_connect_failure( const char *message,
                               int code,
                               const char *code_type );
 
+/**
+ * Raises an error indicating a generic socket failure.
```

Comment:
This is not a generic failure function, it is for when the `socket` function has failed.